protocol/src/index.ts

```diff
@@ -651,6 +651,21 @@ export namespace Protocol {
     '|controlshtml|': readonly ['controlshtml', HTML];
     '|fieldhtml|': readonly ['fieldhtml', HTML];
     '|debug|': readonly ['debug', Message];
+    '|deinit|': readonly ['deinit'];
+    '|selectorhtml|': readonly ['selectorhtml', string, HTML];
+    '|refresh|': readonly ['refresh'];
+    '|tempnotify|': readonly ['tempnotify', string, Message, ...[] | [Message] | [Message, string]];
+    '|tempnotifyoff|': readonly ['tempnotifyoff', string];
+    '|noinit|namerequired|': readonly ['noinit', 'namerequired', Message];
+    '|noinit|nonexistent|': readonly ['noinit', 'nonexistent', Message];
+    '|noinit|joinfailed|': readonly ['noinit', 'joinfailed', Message];
+    '|noinit|rename|': readonly ['noinit', 'rename', string, string];
+    '|hidelines|delete|': readonly ['hidelines', 'delete', string, Num];
+    '|hidelines|hide|': readonly ['hidelines', 'hide', string, Num];
+    '|hidelines|unlink|': readonly ['hidelines', 'unlink', string];
+    '|expire|': readonly ['expire', Message?];
+    '|modaction|': readonly ['modaction', Message];
+    '|askreg|': readonly ['askreg', string];
```

Comment:
`ID`

Reply:
I was initially unsure what `ID` meant or if it's only used in battles. I guess it's also used in user ID's then.

Reply:
`ID` is a pretty fundamental type in PS - its just any all lowercase no non-alpahanumeric string identifier.